src/media-query-provider.js

```diff
@@ -1,68 +1,87 @@
 import React from 'react';
-import matchMedia from 'matchmedia';
 import PropTypes from 'prop-types';
 import shallowequal from 'shallowequal';
-
-const defaultQueries = {
-  mobile: 'screen and (max-width: 623px)',
-  tablet: 'screen and (min-width: 624px) and (max-width: 1020px)',
-  desktop: 'screen and (min-width: 1021px) and (max-width: 1440px)',
-  largeDesktop: 'screen and (min-width: 1441px)',
-};
+// this is for server side rendering and does not use window.matchMedia
+import cssMediaQuery from 'css-mediaquery';
+import isNode from 'detect-node';
 
 class MediaQueryProvider extends React.Component {
-
   constructor(props) {
     super(props);
+
+    const queryTuples = Object.entries(this.props.queries);
```

Comment:
Rather than use `Object.entries` can you use:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/entries#Polyfill
(place this function in another file)

So we dont need to worry about IE support

Reply:
Babel's transform runtime has us covered here, see the dist code requires `babel-runtime/core-js/object/entries`, this is more efficient than library maintainers rolling their own polyfills which will cause logic duplication inside the bundles.